codec-http2/src/main/java/io/netty/handler/codec/http2/HttpConversionUtil.java

```diff
@@ -636,13 +640,22 @@ public void translate(Entry<CharSequence, CharSequence> entry) throws Http2Excep
                 if (COOKIE.equals(name)) {
                     // combine the cookie values into 1 header entry.
                     // https://tools.ietf.org/html/rfc7540#section-8.1.2.5
-                    String existingCookie = output.get(COOKIE);
-                    output.set(COOKIE,
-                               (existingCookie != null) ? (existingCookie + "; " + value) : value);
+                    if (cookies == null) {
+                        cookies = new StringBuilder(value.length());
```

Comment:
I wonder if we should allocate a bit more then `value.length` here to prevent costly "re-allocations" when we have multiple values.

Reply:
it's tricky to guess what the right size should be. i went with adding 512 to value.length but this is arbitrary.

Reply:
Btw did you consider using `InternalThreadLocalMap.stringBuilder()` ?

Reply:
nope. first i've heard of it. looks like it'd be perfect here. will update.

Reply:
TIL. This is great.